From e7dbd17f6a2119b3b38013cf21ec54209214993b Mon Sep 17 00:00:00 2001
From: "ziheng.li" <ziheng.li@amlogic.com>
Date: Thu, 5 Aug 2021 17:35:29 +0800
Subject: [PATCH 4/6] USB: Disable NAK interrupts in ISOC mode [1/1]

PD#SWPL-57028

Problem:
1.When UAC and ADB are enabled at the same time,
the NAK interrupt will be entered continuously.

Solution:
1.Disable NAK interrupts in ISOC mode.

Verify:
A113L T404 Board

Change-Id: I6b3fe6c01501d6c5dea5bc2a684cc080675f1a57
Signed-off-by: ziheng.li <ziheng.li@amlogic.com>
---
 drivers/amlogic/usb/dwc_otg/310/dwc_otg_cil.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/amlogic/usb/dwc_otg/310/dwc_otg_cil.c b/drivers/amlogic/usb/dwc_otg/310/dwc_otg_cil.c
index 60e323800a3d7..80d10edc572e2 100644
--- a/drivers/amlogic/usb/dwc_otg/310/dwc_otg_cil.c
+++ b/drivers/amlogic/usb/dwc_otg/310/dwc_otg_cil.c
@@ -3404,7 +3404,7 @@ void dwc_otg_ep_activate(dwc_otg_core_if_t *core_if, dwc_ep_t *ep)
 				diepmsk.b.intknepmis = 0;
 			diepmsk.b.txfifoundrn = 1;
 			if (ep->type == DWC_OTG_EP_TYPE_ISOC)
-				diepmsk.b.nak = 1;
+				diepmsk.b.nak = 0;
 /*
 			if (core_if->dma_desc_enable) {
 				diepmsk.b.bna = 1;
@@ -3446,7 +3446,7 @@ void dwc_otg_ep_activate(dwc_otg_core_if_t *core_if, dwc_ep_t *ep)
 		if (ep->type == DWC_OTG_EP_TYPE_ISOC) {
 			if (ep->is_in) {
 				diepmsk_data_t diepmsk = {.d32 = 0 };
-				diepmsk.b.nak = 1;
+				diepmsk.b.nak = 0;
 				DWC_MODIFY_REG32(&dev_if->dev_global_regs->diepmsk, 0, diepmsk.d32);
 			} else {
 				doepmsk_data_t doepmsk = {.d32 = 0 };
-- 
2.31.1

